### DISCUSSION TITLE
3.1.2

### DISCUSSION THREAD

**Discussion by LabhanshAgrawal** [vercel/hyper#5826] [Announcements]
V3.1.2

|   OS    | Installer |                                                                                                 |                                                                                                  |
| :-----: | :-------: | :---------------------------------------------------------------------------------------------: | :----------------------------------------------------------------------------------------------: |
|   Mac   |    dmg    | [x64](https://github.com/vercel/hyper/releases/download/v3.1.2/Hyper-3.1.2-mac-x64.dmg) (Intel) | [arm64](https://github.com/vercel/hyper/releases/download/v3.1.2/Hyper-3.1.2-mac-arm64.dmg) (M1) |
|  Linux  |    deb    |      [x64](https://github.com/vercel/hyper/releases/download/v3.1.2/hyper_3.1.2_amd64.deb)      |     [arm64](https://github.com/vercel/hyper/releases/download/v3.1.2/hyper_3.1.2_arm64.deb)      |
|         |    rpm    |     [x64](https://github.com/vercel/hyper/releases/download/v3.1.2/hyper-3.1.2.x86_64.rpm)      |    [arm64](https://github.com/vercel/hyper/releases/download/v3.1.2/hyper-3.1.2.aarch64.rpm)     |
|         | AppImage  |      [x64](https://github.com/vercel/hyper/releases/download/v3.1.2/Hyper-3.1.2.AppImage)       |   [arm64](https://github.com/vercel/hyper/releases/download/v3.1.2/Hyper-3.1.2-arm64.AppImage)   |
|         |   snap    |     [x64](https://github.com/vercel/hyper/releases/download/v3.1.2/hyper_3.1.2_amd64.snap)      |                                                                                                  |
| Windows |    exe    |      [x64](https://github.com/vercel/hyper/releases/download/v3.1.2/Hyper-Setup-3.1.2.exe)      |                                                                                                  |

## Changelog
- Add `disableAutoUpdates` config option (#5805)
- Redirect exec calls from renderer to main process (#5803) (To help with performance issues on macOS while using `child_process` in Renderer)
- Fix file/directory drag and drop (#5781)
- Fix file/directory paste (#5780)
- Add support for updating to arm builds (#5787)

<hr /><em>This discussion was created from the release <a href='https://github.com/vercel/hyper/releases/tag/v3.1.2'>3.1.2</a>.</em>

**Comment by bfaulk96** [comment]
> Redirect exec calls from renderer to main process (redirect exec calls from renderer to main process #5803) (To help with performance issues on macOS while using child_process in Renderer)


Does anyone know if this completely fixes the lagging with hyperline, or just reduces it?
I have a perfectly smooth experience on MacOS Big Sur with `Hyper 3.1.0-canary.6 (stable)` (the test release that reverted to Electron 11) and want to be sure I'm not taking a performance hit before I abandon the canary release for this one.

**Reply by LabhanshAgrawal** [reply]
I'd done a rough comparison using developer tools performance tab, it seemed completely fixed to me.
You don't have to update your current install to test the release. You can download 3.1.2 and run it directly without replacing the canary build. And then switch to it if you find it good.

**Reply by bfaulk96** [reply]
Awesome, thank you!

**Reply by bfaulk96** [reply]
Unfortunately, `3.1.2` is still extremely laggy with hyperline enabled on MacOS Big Sur (2019 w/ Intel Chip)
